### ISSUE TITLE
--disable-uncorrectable does not add comment for Style/ClassAndModuleChildren

### ISSUE THREAD

**Issue by DylanGriffith** [rubocop-hq/rubocop#7621]
## Expected behavior

When running:

```
rubocop --safe-auto-correct --disable-uncorrectable --only 'Style/ClassAndModuleChildren'
```

I should expect that any class definition using the `compact` syntax would end up with a comment added disabling this cop.

## Actual behavior

The file is unchanged and it still produces an error.

## Steps to reproduce the problem

```bash
$ cat foo_bar.rb
# frozen_string_literal: true

class Foo::Bar
end

$ rubocop --safe-auto-correct --disable-uncorrectable --only 'Style/ClassAndModuleChildren'
Inspecting 1 file
C

Offenses:

foo_bar.rb:3:7: C: Style/ClassAndModuleChildren: Use nested module/class definitions instead of compact style.
class Foo::Bar
      ^^^^^^^^

1 file inspected, 1 offense detected

$ cat foo_bar.rb
# frozen_string_literal: true

class Foo::Bar
end
```

Both `--auto-correct` and `--safe-auto-correct` seem to have the same behaviour here. I was able to get it to correct this by configuring `AutoCorrect: true` for this cop but I don't want to autocorrect these anyway since it is unsafe.

## RuboCop version

```
$ rubocop -V
0.78.0 (using Parser 2.6.5.0, running on ruby 2.6.3 x86_64-darwin18)
```

**Comment by moger777** [comment]
There seems to be several cops that for whatever reason don't get `# rubocop:todo` for whatever reason. Another example is `Lint/UselessAssignment`.

**Comment by laurmurclar** [comment]
`Style/MultilineBlockChain` also seems to suffer from this

**Comment by patbl** [comment]
At least now, I think the original issue is due to general Rubocop behavior rather than anything specific to this cop: if a cop supports unsafe auto-correct, you can't use `--disable-uncorrectable` to add `rubocop:todo` comments. There's a feature request for this: https://github.com/rubocop/rubocop/issues/7958.
